### PR TITLE
tests/resource/aws_elasticsearch_domain: Add sweeper

### PR DESCRIPTION
**Pull request by bflad**
♻️ 

```
$ aws es list-domain-names | jq '.DomainNames | length'
16

$ make sweep SWEEPARGS='-sweep-run=aws_elasticsearch_domain'
...
2018/03/15 19:46:43 Sweeper Tests ran:
	- aws_elasticsearch_domain
ok  	github.com/terraform-providers/terraform-provider-aws/aws	16.029s

$ aws es list-domain-names | jq '.DomainNames | length'
1
```

### CODE REVIEW

File: aws/resource_aws_elasticsearch_domain_test.go [hunk 1]
```diff
@@ -13,6 +15,55 @@ import (
...
+		}
+		log.Printf("[INFO] Deleting Elasticsearch Domain: %s", *domain.DomainName)
+
+		_, err := conn.DeleteElasticsearchDomain(&elasticsearch.DeleteElasticsearchDomainInput{
+			DomainName: domain.DomainName,
+		})
+		if err != nil {
+			log.Printf("[ERROR] Failed to delete Elasticsearch Domain %s: %s", *domain.DomainName, err)
+			continue
+		}
```
Comment (radeksimko) [inline aws/resource_aws_elasticsearch_domain_test.go]: I think the convention, even for sweepers, is to wait until the resource _actually_ gets deleted, i.e. do you mind decoupling [the waiter](https://github.com/terraform-providers/terraform-provider-aws/blob/3040b33e70bf36fd184664d1d4f1bab1bb14c28b/aws/resource_aws_elasticsearch_domain.go#L689-L714) and using it here?

If we don't do that then we might still run into limits as tests will run when domains are being deleted, yet still consuming the capacity and also the subsequent run of sweepers (post-sweepers) will try to delete those domains (again).

Comment (bflad) [reply aws/resource_aws_elasticsearch_domain_test.go]: No problem, I'll add that in

**Comment by bflad** [comment]
Updated!

```
15 tests passed (all tests)
=== RUN   TestAccAWSElasticSearchDomain_duplicate
--- PASS: TestAccAWSElasticSearchDomain_duplicate (524.75s)
=== RUN   TestAccAWSElasticSearchDomain_importBasic
--- PASS: TestAccAWSElasticSearchDomain_importBasic (668.49s)
=== RUN   TestAccAWSElasticSearchDomain_encrypt_at_rest_default_key
--- PASS: TestAccAWSElasticSearchDomain_encrypt_at_rest_default_key (915.58s)
=== RUN   TestAccAWSElasticSearchDomain_v23
--- PASS: TestAccAWSElasticSearchDomain_v23 (964.78s)
=== RUN   TestAccAWSElasticSearchDomain_complex
--- PASS: TestAccAWSElasticSearchDomain_complex (975.05s)
=== RUN   TestAccAWSElasticSearchDomain_policy
--- PASS: TestAccAWSElasticSearchDomain_policy (976.46s)
=== RUN   TestAccAWSElasticSearchDomain_encrypt_at_rest_specify_key
--- PASS: TestAccAWSElasticSearchDomain_encrypt_at_rest_specify_key (988.00s)
=== RUN   TestAccAWSElasticSearchDomain_LogPublishingOptions
--- PASS: TestAccAWSElasticSearchDomain_LogPublishingOptions (1092.43s)
=== RUN   TestAccAWSElasticSearchDomain_vpc
--- PASS: TestAccAWSElasticSearchDomain_vpc (1147.94s)
=== RUN   TestAccAWSElasticSearchDomain_tags
--- PASS: TestAccAWSElasticSearchDomain_tags (1272.62s)
=== RUN   TestAccAWSElasticSearchDomainPolicy_basic
--- PASS: TestAccAWSElasticSearchDomainPolicy_basic (1435.15s)
=== RUN   TestAccAWSElasticSearchDomain_update
--- PASS: TestAccAWSElasticSearchDomain_update (1967.81s)
=== RUN   TestAccAWSElasticSearchDomain_basic
--- PASS: TestAccAWSElasticSearchDomain_basic (2020.23s)
=== RUN   TestAccAWSElasticSearchDomain_vpc_update
--- PASS: TestAccAWSElasticSearchDomain_vpc_update (2513.81s)
=== RUN   TestAccAWSElasticSearchDomain_internetToVpcEndpoint
--- PASS: TestAccAWSElasticSearchDomain_internetToVpcEndpoint (2578.69s)
```

File: aws/resource_aws_elasticsearch_domain_test.go [hunk 2]
```diff
@@ -13,6 +15,56 @@ import (
...
+		resourceAwsElasticSearchDomainDeleteWaiter(*domain.DomainName, conn)
```
Comment (radeksimko) [inline aws/resource_aws_elasticsearch_domain_test.go]: Nitpick: I think we should also catch the error here.

Comment (bflad) [reply aws/resource_aws_elasticsearch_domain_test.go]: Ask and you shall receive. 😄 
```
		err = resourceAwsElasticSearchDomainDeleteWaiter(*domain.DomainName, conn)
		if err != nil {
			log.Printf("[ERROR] Failed to wait for deletion of Elasticsearch Domain %s: %s", *domain.DomainName, err)
		}
```

**Pull request merged**